### PR TITLE
[BISERVER-12601] MS SQL repository: csv data source can't be created if any of column has non zero precision

### PR DESCRIPTION
**Pull request by IvanNikolaychuk**
- In case of non zero precision and zero length choosing float as a type for a column.
- Tests wirtten.
- Checkstyle applied.

@brosander, could you please review it?

### CODE REVIEW

**Comment by pamval** [comment]
running wingman

**Comment by lucboudreau** [comment]
Build Commands:
==============
Build:
------
1. rm -r ~/.kettle; rm -r ~/.pentaho; echo removed hidden folders
2. ant -Djunit.jvmargs=-XX:PermSize=256m -f core/build.xml clean-all resolve jacoco jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  core/src/org/pentaho/di/core/database/MSSQLServerDatabaseMeta.java
  core/test-src/org/pentaho/di/core/database/MSSQLServerDatabaseMeta_FieldDefinitionTest.java


Checkstyle violations:
======================
See: https://github.com/pentaho/pentaho-coding-standards

File core/test-src/org/pentaho/di/core/database/MSSQLServerDatabaseMeta_FieldDefinitionTest.java
1. Message: Using a static member import should be avoided - junit.framework.Assert.assertEquals. at Line: 32 Column: None
2. Message: Using a static member import should be avoided - junit.framework.Assert.assertTrue. at Line: 33 Column: None









Unit test coverage change:
============
org.pentaho.di.core.database.MSSQLServerDatabaseMeta:
 * branchPercentCoverage :29.23%
 * instructionPercentCoverageChange : +6.58%
 * linePercentCoverageChange : +20.62%
 * complexityPercentCoverageChange : +12.86%
 * methodPercentCoverageChange : +5.56%

**Comment by pamval** [comment]
@akhayrutdinov review please

File: core/test-src/org/pentaho/di/core/database/MSSQLServerDatabaseMeta_FieldDefinitionTest.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+import org.pentaho.di.core.row.ValueMetaInterface;
+import org.pentaho.di.core.row.value.ValueMetaNone;
+import org.pentaho.di.core.row.value.ValueMetaNumber;
+import org.pentaho.di.core.row.value.ValueMetaString;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertTrue;
+
+public class MSSQLServerDatabaseMeta_FieldDefinitionTest {
+  private static MSSQLServerDatabaseMeta dbMeta;
```
Comment (akhayrutdinov) [inline core/test-src/org/pentaho/di/core/database/MSSQLServerDatabaseMeta_FieldDefinitionTest.java]: `static`?

File: core/test-src/org/pentaho/di/core/database/MSSQLServerDatabaseMeta_FieldDefinitionTest.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    ValueMetaInterface valueMeta = new MetaInterfaceBuilder( ValueMetaInterface.TYPE_STRING )
+      .length( 10_000 )
+      .build();
+
+
+    assertEquals( STRING_TEXT, dbMeta.getFieldDefinition( valueMeta, null, null, false, false, false ) );
+
+  }
+
+  private class MetaInterfaceBuilder {
```
Comment (akhayrutdinov) [inline core/test-src/org/pentaho/di/core/database/MSSQLServerDatabaseMeta_FieldDefinitionTest.java]: that's is not essential for testing, but I doubt the class should be non-static

**Comment by akhayrutdinov** [comment]
The rest is ok for me

**Pull request merged**